yarn-project/aztec-cli/src/index.ts

```diff
@@ -103,11 +103,13 @@ export function getProgram(log: LogFn, debugLogger: DebugLogger): Command {
       let publicKey;
       if (options.mnemonic) {
         const acc = mnemonicToAccount(options.mnemonic);
-        const key = Buffer.from(acc.getHdKey().privateKey!);
-        privKey = key.toString('hex');
-        publicKey = await generatePublicKey(new PrivateKey(key));
+        // Given that the mnemonicToAccount is a function from viem/accounts, it works with Ethereum private keys.
+        // Ethereum private keys are 256 bits and for this reason we need to fit it to Fr as that is the canonical
+        // representation of a private key used with Grumpkin curve.
```

Comment:
Is there a typo here? My understanding is that Fr is the field used in Noir which is different to the canonical representation of the field used by grumpkin. 

Also not sure I get the link between ethereum private keys and Fr, since those use two different prime fields, there will not be bijection, this may be fine depending on how they are used; the consequence being that some keys have more than one representation when mapped to the other field

Reply:
> Is there a typo here? My understanding is that Fr is the field used in Noir which is different to the canonical representation of the field used by grumpkin.

It's not a typo, it's my lack of knowledge 😏 

Based on [this](https://noir-lang.org/language_concepts/data_types/#fields) I assumed that `Fr` is the canonical representation of the field used by Grumpkin. In the [Aztec Yellow Paper](https://hackmd.io/@aztec-network/ByzgNxBfd) it says that the and field order of grumpkin is the same as BN-254:
<img width="779" alt="image" src="https://github.com/AztecProtocol/aztec-packages/assets/13470840/b5dcb7b8-108f-4a9c-a13f-be1e0423e510">

@kevaundray what modulus should then be used for the canonical representation of field used by Grumpkin?

> Also not sure I get the link between ethereum private keys and Fr, since those use two different prime fields, there will not be bijection, this may be fine depending on how they are used; the consequence being that some keys have more than one representation when mapped to the other field

We don't need a bijection here, we simply need to generate a private key from a mnemonic which will be used for note encryption. Since we have the viem library available, we generate Ethereum key first and then we reduce that to the canonical representation of the grumpkin one (At least that's what I though so before learning that Fr is not that). I will need to clarify this in the CLI because I can imagine that it confuses people if the private key we output here is different from the one outputted by BIP-39.